pkg/operation/botanist/kubeapiserver.go

```diff
@@ -198,29 +201,49 @@ func (b *Botanist) ensureAdmissionPluginConfig(plugins []gardencorev1beta1.Admis
 		}
 	}
 
+	if index == -1 {
+		return plugins, nil
+	}
+
 	// If user has set a config in the shoot spec, retrieve it
 	if plugins[index].Config != nil {
+		var (
+			admissionConfigData []byte
+			err                 error
+		)
+
 		config, err := runtime.Decode(codec, plugins[index].Config.Raw)
 		if err != nil {
 			return nil, err
 		}
-		admissionConfig, ok := config.(*admissionapiv1beta1.PodSecurityConfiguration)
-		if !ok {
-			return nil, fmt.Errorf("expected admissionapiv1beta1.PodSecurityConfiguration but got %T", config)
-		}
 
 		// Add kube-system to exempted namespaces
-		if !slices.Contains(admissionConfig.Exemptions.Namespaces, metav1.NamespaceSystem) {
-			admissionConfig.Exemptions.Namespaces = append(admissionConfig.Exemptions.Namespaces, metav1.NamespaceSystem)
+		if _, ok := config.(*admissionapiv1alpha1.PodSecurityConfiguration); ok {
```

Comment:
Using type switches is the "go" way to check if an interface contains a particular type:
```
switch admissionConfig := config.(type) {
case *admissionapiv1alpha1.PodSecurityConfiguration:
        if !slices.Contains(admissionConfig.Exemptions.Namespaces, metav1.NamespaceSystem) {
            admissionConfig.Exemptions.Namespaces = append(admissionConfig.Exemptions.Namespaces, metav1.NamespaceSystem)
	}
	admissionConfigData, err = runtime.Encode(codec, admissionConfig)

case *admissionapiv1beta1.PodSecurityConfiguration:
...
default: 
	err = fmt.Errorf("expected PodSecurityConfiguration but got %T", config)
}